deepchem/data/dft_data.py

```diff
@@ -0,0 +1,319 @@
+        self._systems = [DFTSystem.create(p) for p in entry_dct["systems"]]
+        self._dtype = dtype
+        self._device = device
+
+        self._trueval_is_set = False
+        self._trueval = torch.tensor(0.0).to(device)
+        """
+        Parameters
+        ----------
+        systems:  List[DFTSystem]
+            Returns the list of systems in the entry
+        dtype: torch.dtype
+            Returns data type of the calculated and true energies of a system
+        device: torch.device
+        """
+
+    @property
+    def dtype(self) -> torch.dtype:
+        return self._dtype
+
+    @property
+    def device(self) -> torch.device:
+        return self._device
+
+    def get_systems(self) -> List[DFTSystem]:
+        """
+        Returns the list of systems in the entry
+        """
+        return self._systems
+
+    @abstractproperty
+    def entry_type(self) -> str:
+        """
+        Returning the type of the entry of the dataset; 
+        1) Atomic Ionization Potential (IP/IE)
+        2) Atomization Energy (AE)
+        3) Density Profile (DENS)
+        4) Density Matrix (DM)
+        """
+        pass
+
+    def get_true_val(self) -> torch.Tensor:
+        if not self._trueval_is_set:
+            self._trueval = self._get_true_val()
+            self._trueval_is_set = True
+        return self._trueval
+
+    @abstractmethod
+    def _get_true_val(self) -> torch.Tensor:
+        """
+        Get the true value of the DFTEntry.  
+        For the AE and IP entry types, the experimental values are collected from the NIST CCCBDB/ASD 
+        databases. 
+        The true values of density profiles are calculated using PYSCF-CCSD calculations. This method            simply loads the value, no calculation is performed.  
+        """
+        pass
+
+    @abstractmethod
+    def get_val(self, qcs: List[KSCalc]) -> torch.Tensor:
+        """
+        Return the energy value of the entry, using a DQC-DFT calculation, where the XC has been 
+        replaced by the trained neural network. This method does not carry out any calculations, it is 
+        an interface to the KSCalc utility. 
+        """
+        pass
+
+
+class EntryDM(DFTEntry):
+    """Entry for Density Matrix (DM)"""
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        assert len(self.get_systems()) == 1, "dm entry can only have 1 system"
+
+    @property
+    def entry_type(self) -> str:
+        return "dm"
+
+    def _get_true_val(self) -> torch.Tensor:
+        # get the density matrix from PySCF's CCSD calculation
+        dm = np.load(self["true_val"])
+        true_val = torch.from_numpy(dm)
+        return torch.as_tensor(true_val, device=self.device)
+
+    def get_val(self, qcs: List[KSCalc]) -> torch.Tensor:
+        return qcs[0].aodmtot()
+
+
+class EntryDens(DFTEntry):
+    """Entry for density profile (dens), compared with CCSD calculation"""
+
+    def __init__(self, *args, **kwargs):
```

Comment:
What are the allowed values of args/kwargs? In general, it's not good to have *args, **kwargs in a class definition. It means that the person reading the code doesn't know the arguments or key word arguments they need to use with the class